src/main/scala/org/ergoplatform/nodeView/wallet/persistence/Balance.scala

```diff
@@ -1,13 +1,13 @@
 package org.ergoplatform.nodeView.wallet.persistence
 
-import org.ergoplatform.nodeView.wallet.IdUtils._
+import org.ergoplatform.ErgoBox.{BoxId, TokenId}
 import org.ergoplatform.wallet.boxes.TrackedBox
 
-case class Balance(id: EncodedBoxId,
+case class Balance(id: BoxId,
                    value: Long,
-                   assets: Map[EncodedTokenId, Long])
+                   assets: Map[TokenId, Long])
```

Comment:
Yes I am aware that arrays don't work as keys because there is no specialized hash function, but they can still store the values. They get converted to strings when Balances are merged [here.](https://github.com/ergoplatform/ergo/pull/1975/files#diff-80148921533d33364e636b1a485685bffdaca1d0d5932afc303ad5774f96b918R35)

Reply:
The problem is in comparison, not hashing. My example shows exactly that. Comparison for arrays is done by-reference, not by-value, so you will not get a record for the same key bytes (you will get it only if you provide the same array)

Reply:
Resolved with a .toList

Reply:
Please convert to ModifierId instead, this is standard way to avoid issues with Array which was agreed on after different coders tried different approaches (WrappedArray / ByteArrayWrapper etc)

Reply:
Changed to ModifierId